meta-refkit-core/recipes-images/images/refkit-installer-image.bb

```diff
@@ -63,39 +63,89 @@ REFKIT_INSTALLER_UEFI_COMBO () {
-                # We store a random key in the TPM NVRAM where it is accessible
-                # to the initramfs. The initramfs will turn off read-access
-                # after it has retrieved the key, so nothing else that gets started
-                # later will have access to the key.
-                if ! execute tpm_nvdefine -i "${REFKIT_DISK_ENCRYPTION_NVRAM_INDEX}" -s "$( expr "${REFKIT_DISK_ENCRYPTION_NVRAM_ID_LEN}" + "${REFKIT_DISK_ENCRYPTION_KEY_SIZE}" )" -p 'AUTHREAD|AUTHWRITE|READ_STCLEAR' -y -z; then
-                    fatal "creating NVRAM area failed"
-                fi
-                if ! (printf "%s" "${REFKIT_DISK_ENCRYPTION_NVRAM_ID}" &&
-                      dd if=/dev/urandom bs="${REFKIT_DISK_ENCRYPTION_KEY_SIZE}" count=1) >"$keyfile"; then
-                    fatal "key creation failed"
-                fi
-                keyfile_offset="${REFKIT_DISK_ENCRYPTION_NVRAM_ID_LEN}"
-                if ! execute tpm_nvwrite -i "${REFKIT_DISK_ENCRYPTION_NVRAM_INDEX}" -z -f "$keyfile"; then
-                    fatal "storing key in NVRAM failed"
-                fi
-                # Lock access until reboot.
-                if ! execute tpm_nvread -i "${REFKIT_DISK_ENCRYPTION_NVRAM_INDEX}" -z -s 0; then
-                    fatal "locking key in NVRAM failed"
-                fi
-            fi
+            # explicitly by setting the TPM env variable to "1.2", "2.0", or "no".
+            # The default is "no".
+            TPM=${TPM:-no}
+            case "$TPM" in
+                2.0)
+                    if ! ${@ bb.utils.contains('DISTRO_FEATURES', 'tpm2', 'true', 'false', d) }; then
+                        fatal "support for TPM 2.0 is not enabled"
+                    fi
+
+                    # This tells tpm2.0-tools that we want to access the /dev/tpm0 device.
+                    # Setting these environment variables works with tpm2.0-tools 2.x and 3.x,
+                    # whereas command line parameters changed between that.
+                    TPM2TOOLS_TCTI_NAME=device
+                    TPM2TOOLS_DEVICE_FILE=/dev/tpm0
+                    export TPM2TOOLS_TCTI_NAME TPM2TOOLS_DEVICE_FILE
+
+                    # NVRAM can be used with TPM 2.0 without taking ownership, so we skip that step.
+                    #
+                    # We store a random key in the TPM NVRAM where it is accessible
+                    # to the initramfs. The initramfs will turn off read-access
+                    # after it has retrieved the key, so nothing else that gets started
+                    # later will have access to the key.
+                    #
+                    # -t 0x80020002 = TPMA_NV_OWNERWRITE|TPMA_NV_OWNERREAD|TPMA_NV_READ_STCLEAR
+                    # (https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf)
+                    #
+                    # tpm2.0-tools master supports --attibutes=ownerwrite|ownerread|read_stdclear, but
```

Comment:
Just for my own education: is master same as road to 3.x (referenced by the initramfs code)? Is that going to be available soon?

Reply:
There is now a 3.0rc1 based on master, so 3.0 might not be too far away.

Reply:
Thanks!